casper/src/main/scala/coop/rchain/casper/merging/DeployIndex.scala

```diff
@@ -3,18 +3,31 @@ package coop.rchain.casper.merging
 import cats.effect.Concurrent
 import cats.syntax.all._
 import com.google.protobuf.ByteString
-import coop.rchain.casper.protocol.ProcessedDeploy
+import coop.rchain.casper.protocol.Event
 import coop.rchain.rspace.merger.EventLogIndex
 
 /** index of a single deploy */
-final case class DeployIndex(deployId: ByteString, cost: Long, eventLogIndex: EventLogIndex)
+final case class DeployIndex(
+    deployId: ByteString,
+    cost: Long,
+    eventLogIndex: EventLogIndex
+)
 
 object DeployIndex {
+  // This cost is required because rejection option selection rule depends on how much branch costs.
+  // For now system deploys do not have any weight, cost is 0.
+  val SYS_SLASH_DEPLOY_COST       = 0L
+  val SYS_CLOSE_BLOCK_DEPLOY_COST = 0L
+  val SYS_EMPTY_DEPLOY_COST       = 0L
+  // These are to be put in rejected set in blocks, so prefix format is defined for identification purposes.
+  val SYS_SLASH_DEPLOY_ID       = ByteString.copyFrom(64.toByte +: new Array[Byte](31))  // 1000000xxx
```

Comment:
Maybe we should use something else like blockHash instead of fixed id?

I am thinking in the future we can optimize merging by rejecting with the same deployId directly without detailed conflict detections.

Reply:
I'm not sure, would be nice to be able to identify particular deploy. Maybe having block hash with some prefix? I'm not sure though if its ok to have different length.